NixOS/nixpkgs
# mjolnir: init at v0.1.17

###### Motivation for this change

https://github.com/matrix-org/mjolnir

The #123879 PR only added the Synapse module. This PR packages the core of the bot.

###### Manual Testing Performed

This is the trace of execution for mjolnir. Note that I haven't configured it properly, hence the crash, but note that there are a few info lines before the error indicating that the bot was initialized successfully.

```
$ NODE_CONFIG_DIR=/home/sumner/tmp ./result/bin/mjolnir
Sat, 29 May 2021 07:55:09 GMT [INFO] [index] Starting bot...
Sat, 29 May 2021 07:55:09 GMT [INFO] [MatrixLiteClient (REQ-1)] GET https://matrix.sumnerevans.com/_matrix/client/r0/joined_rooms
Sat, 29 May 2021 07:55:09 GMT [INFO] [index] Resolving protected rooms...
Sat, 29 May 2021 07:55:09 GMT [INFO] [index] Resolving management room...
Sat, 29 May 2021 07:55:09 GMT [INFO] [MatrixLiteClient (REQ-2)] GET https://matrix.sumnerevans.com/_matrix/client/r0/directory/room/%23moderators%3Asumnerevans.org
Sat, 29 May 2021 07:55:10 GMT [ERROR] [MatrixLiteClient (REQ-2)] { errcode: 'M_UNKNOWN', error: 'Internal server error' }
(node:924623) UnhandledPromiseRejectionWarning: #<IncomingMessage>
(Use `node --trace-warnings ...` to show where the warning was created)
(node:924623) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:924623) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

###### Things done



- Built on platform(s)
- [ ] Added a release notes entry if the change is major or breaking
